client/consensus/relay-chain/src/import_queue.rs

```diff
@@ -58,6 +58,18 @@ where
 		&mut self,
 		mut block_params: BlockImportParams<Block, ()>,
 	) -> Result<(BlockImportParams<Block, ()>, Option<Vec<(CacheKeyId, Vec<u8>)>>), String> {
+		if block_params.with_state() {
```

Comment:
Super nit (feel free to ignore). In [this](https://github.com/paritytech/substrate/pull/13221) new PR we've grouped the checks together. If you like we can do the same here .